### DISCUSSION TITLE
Set up timeout session in Superset application for UI users.

### DISCUSSION THREAD

**Discussion by kvdevopse** [apache/superset#10980] [General]
Set up timeout session for users in Superset Application
 
Hi, We set up the superset application and we are trying to achieve the application UI timeout session for the users. In config.py file we setup "PERMANENT_SESSION_LIFETIME = 60" for one minute but session timeout for users is not working. The logout for users is not happening automatically and also if we refresh the superset application UI page, it is logging in directly without prompting login page.

These are the timeout session values we passed in config file.
SUPERSET_WEBSERVER_TIMEOUT = 60
**PERMANENT_SESSION_LIFETIME = 60**

SQLLAB_TIMEOUT = 30

Please help us to fix the timeout session.

**Comment by shawnzhu** [comment]
See example from [Flask app Configuration Hook](https://superset.apache.org/docs/installation/configuring-superset#flask-app-configuration-hook) and details here https://github.com/apache/superset/pull/16286
